packages/frontend/src/components/TableConfigPanel/ConditionalFormatting.tsx

```diff
@@ -0,0 +1,211 @@
+
+const ConditionalFormatting: FC = () => {
+    const {
+        explore,
+        tableConfig: { conditionalFormattings, onSetConditionalFormattings },
+    } = useVisualizationContext();
+
+    const activeFields = useExplorerContext((c) => c.state.activeFields);
+    const visibleActiveNumericFields = useMemo(() => {
+        return explore
+            ? getVisibleFields(explore)
+                  .filter((field) => activeFields.has(fieldId(field)))
+                  .filter((field) => field.type === 'number')
+            : [];
+    }, [explore, activeFields]);
+
+    // TODO: this is intentional to support multiple conditional formattings
+    const conditionalFormatting = conditionalFormattings[0];
+    const field = useMemo(
+        () =>
+            visibleActiveNumericFields.find(
+                (f) => getItemId(f) === conditionalFormatting?.target?.fieldId,
+            ),
+        [conditionalFormatting, visibleActiveNumericFields],
+    );
+
+    const [config, setConfig] = useState<ConditionalFormattingConfig | null>(
+        conditionalFormatting,
+    );
+
+    const handleAddEmptyConditionalFormatting = () => {
+        setConfig({
+            target: null,
+            color: '#BBAAFF',
+            rules: [createConditionalFormatingRule()],
+        });
+    };
+
+    const handleRemoveConditionalFormatting = () => {
+        setConfig(null);
+    };
+
+    const handleChangeField = (newField: CompiledField | undefined) => {
+        if (!config) return;
+
+        setConfig({
+            ...config,
+            target: newField ? { fieldId: getItemId(newField) } : null,
+        });
+    };
+
+    const handleChangeFilterOperator = (newOperator: FilterOperator) => {
+        if (!config) return;
+
+        setConfig(
+            produce(config, (draft) => {
+                draft.rules[0].operator = newOperator;
+            }),
+        );
+    };
+
+    const handleChangeRule = (newRule: ConditionalFormattingRule) => {
+        if (!config) return;
+
+        setConfig(
+            produce(config, (draft) => {
+                // TODO: this is intentional to support multiple rules in the future
+                draft.rules[0] = newRule;
+                // TODO: check if we can fix this problem in number input
+                draft.rules[0].values = draft.rules[0].values.map((v) =>
+                    Number(v),
+                );
+                return draft;
+            }),
+        );
+    };
+
+    const handleChangeColor = (newColor: string) => {
+        if (!config) return;
+
+        setConfig({ ...config, color: newColor });
+    };
+
+    useEffect(() => {
+        onSetConditionalFormattings(config ? [config] : []);
+    }, [config, onSetConditionalFormattings]);
+
+    // conditional formatting only supports number fields for now
+    const filterConfig = FilterTypeConfig[FilterType.NUMBER];
+
+    // TODO: remove nonsense props
```

Comment:
TODO

Reply:
oops, not relevant anymore, forgot to remove the comment